test/extended/pods/pdb.go

```diff
@@ -0,0 +1,182 @@
+		if err != nil {
+			e2e.Failf("unable to get names of static pods: %v", err)
+		}
+		// knownBugs := map[string]string{} // Create bugs in future
+		pdbMissingWorkloads := []string{}
+		pdbMisconfiguredWorkloads := []string{}
+		operatorsHavingPDBS := []string{}
+		for _, workload := range items {
+			var meta *metav1.ObjectMeta
+			var selectorLabels *metav1.LabelSelector
+			var replicas int32
+			switch t := workload.(type) {
+			case *appsv1.Deployment:
+				t.Kind = "Deployment"
+				meta = &t.ObjectMeta
+				selectorLabels = t.Spec.Selector
+				replicas = t.Status.Replicas
+			case *appsv1.StatefulSet:
+				t.Kind = "StatefulSet"
+				meta = &t.ObjectMeta
+				selectorLabels = t.Spec.Selector
+				replicas = t.Status.Replicas
+			case *appsv1.DaemonSet:
+				t.Kind = "DaemonSet"
+				meta = &t.ObjectMeta
+				selectorLabels = t.Spec.Selector
+				replicas = t.Status.DesiredNumberScheduled
+			default:
+				panic("not an object")
+			}
+			if !strings.Contains(meta.Namespace, "openshift-") {
+				// Not every component is following the convention of -operator at the end, there are some namespaces
+				// like openshift-monitoring and openshift-network-diagnostics and some of them run on worker nodes
+				// as well. So, need to figure out how to exclude operator components and components running on worker node
+				continue
+			}
+
+			key := fmt.Sprintf("%s/%s/%s", workload.GetObjectKind().GroupVersionKind().Kind, meta.Namespace, meta.Name)
+			var pdbs *v1beta1.PodDisruptionBudgetList
+			isPDBConfigured := false
+			var currentPDB v1beta1.PodDisruptionBudget
+			pdbs, err = kubeClient.PolicyV1beta1().PodDisruptionBudgets(meta.Namespace).List(context.TODO(), metav1.ListOptions{})
+			if err != nil {
+				e2e.Failf("Unexpected error getting PDBs: %v", err)
+			}
+			for _, pdb := range pdbs.Items {
+				if reflect.DeepEqual(pdb.Spec.Selector, selectorLabels) {
+					isPDBConfigured = true
+					currentPDB = pdb
+					operatorsHavingPDBS = append(operatorsHavingPDBS, key)
+					// The convention we came up with for static pod control plane components is to have a deployment
+					// doing health check for the corresponding component. This can be changed to labels later.
+					// As of now, I am assuming if the deployment name has `-guard` in the name and it runs in
+					// `openshift-*` namespace, it can be considered a deployment guarding control plane static pod
+					if strings.Contains(meta.Name, "-guard") &&
+						controlPlaneStaticPodNames.Has(meta.Name[:strings.LastIndex(meta.Name, "-")]) {
+						e2e.Logf("The workload %s is having a PDB %s to monitor the health of control plane component %s",
+							key, pdb.Name, meta.Name[:strings.LastIndex(meta.Name, "-")])
+					} else {
+						e2e.Logf("Workload %s has PDB %s configured", key, pdb.Name)
+					}
+					break
+				}
+			}
+			if inHAMode {
+				key = "HA/" + key
+				if !isPDBConfigured {
+					pdbMissingWorkloads = append(pdbMissingWorkloads, key)
```

Comment:
I don't think we need to complain about missing PDBs on `replicas=1` workloads; they're clearly interruptible.  Can we restructure this like:

```go
if replicas > 1 {  // no reason to set this unless we are trying to be HA, right?
  if currentPDB == nil { // you'd need to make currentPDB a pointer for this to work
    pdbMissingWorkloads = append(pdbMissingWorkloads, key)
  } else {
    // checks to see if we were comfortable with the PDB configuration
  }
}
```

Reply:
I can change it. In fact, I can ignore it before the loop